chessmate/engines.py

```diff
@@ -202,8 +203,32 @@ def evaluate(self, board: chess.Board) -> None:
             self.evaluation_function.evaluate(board)
         )
 
+    def move(self, board: chess.Board) -> chess.Move:
```

Comment:
I like this setup. Would you be able to make wrap these prioritize engines into a more generic engine i.e PrioritizePieceMoves where one of the init arguments is the piece symbol for the prioritized piece? For example, PrioritizePieceMoves(piece='pawn') would be the equivalent of the current PrioritizePawnMoves.